CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java

```diff
@@ -387,37 +434,56 @@ private static int processTransactions(ImportDefinition definition, GraphWriteMe
-                }
-                if (initialiseWithSchema && graph.getSchema() != null) {
-                    graph.getSchema().completeVertex(graph, sourceVertexId);
-                }
 
-                final int destinationVertexId = graph.addVertex();
-                for (final ImportAttributeDefinition attributeDefinition : destinationVertexDefinitions) {
-                    attributeDefinition.setValue(graph, destinationVertexId, row, (i - 1));
-                }
-                if (initialiseWithSchema && graph.getSchema() != null) {
-                    graph.getSchema().completeVertex(graph, destinationVertexId);
-                }
+                try {
+                    final int sourceVertexId = graph.addVertex();
+                    final int destinationVertexId = graph.addVertex();
+                    final boolean isDirected = directedIx == ImportConstants.ATTRIBUTE_NOT_ASSIGNED_TO_COLUMN || Boolean.parseBoolean(row[directedIx]);
+                    final int transactionId = graph.addTransaction(sourceVertexId, destinationVertexId, isDirected);
 
-                final boolean isDirected = directedIx == ImportConstants.ATTRIBUTE_NOT_ASSIGNED_TO_COLUMN || Boolean.parseBoolean(row[directedIx]);
-                final int transactionId = graph.addTransaction(sourceVertexId, destinationVertexId, isDirected);
-                for (final ImportAttributeDefinition attributeDefinition : transactionDefinitions) {
-                    if (attributeDefinition.getOverriddenAttributeId() != Graph.NOT_FOUND) {
-                        attributeDefinition.setValue(graph, transactionId, row, (i - 1));
+                    for (final ImportAttributeDefinition attributeDefinition : transactionDefinitions) {
+                        if (attributeDefinition.getOverriddenAttributeId() != Graph.NOT_FOUND) {
+                            attributeDefinition.setValue(graph, transactionId, row, (i - 1));
+                        }
+                    }
+
+                    for (final ImportAttributeDefinition attributeDefinition : destinationVertexDefinitions) {
+                        attributeDefinition.setValue(graph, destinationVertexId, row, (i - 1));
+                    }
+
+                    for (final ImportAttributeDefinition attributeDefinition : sourceVertexDefinitions) {
+                        attributeDefinition.setValue(graph, sourceVertexId, row, (i - 1));
+                    }
+                    if (initialiseWithSchema && graph.getSchema() != null) {
+                        graph.getSchema().completeVertex(graph, sourceVertexId);
+                    }
+
+                    if (initialiseWithSchema && graph.getSchema() != null) {
+                        graph.getSchema().completeVertex(graph, destinationVertexId);
+                    }
+
+                    if (initialiseWithSchema && graph.getSchema() != null) {
+                        graph.getSchema().completeTransaction(graph, transactionId);
```

Comment:
These 3 if conditions are all the same so they can be combined into one if they're all correct

Reply:
Done